producer/src/blockchain/blockchain.go

```diff
@@ -33,30 +34,51 @@ func AddBlock(b block.Block, filename string, databaseName string) error { // Ad
 		return err
 	}
 
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return err
+	}
+	bPosition := fileInfo.Size()
+
 	serialized := b.Serialize()
 	bLen := len(serialized)
 	payload := make([]byte, 4)
 	binary.LittleEndian.PutUint32(payload, uint32(bLen))
 	payload = append(payload, serialized...)
+
 	if _, err := file.Write(payload); err != nil {
 		fmt.Println(err)
 		return err
 	}
+
 	if err := file.Close(); err != nil {
 		log.Fatalln(err)
 	}
 
-	// database, err := sql.Open("sqlite3i", databaseName)
-	// // Checks if the opening was successful
-	// if err != nil {
-	//     return err
-	// }
+	database, err := sql.Open("sqlite3", databaseName)
+	// Checks if the opening was successful
+	if err != nil {
+		return err
+	}
 
-	// statement, _ := database.Prepare("CREATE TABLE IF NOT EXISTS metadata (height INTEGER PRIMARY KEY, position INTEGER, size INTEGER, hash TEXT)")
-	// statement.Exec()
-	// statement, _ = database.Prepare("INSERT INTO")
+	statement, err := database.Prepare("CREATE TABLE IF NOT EXISTS metadata (height INTEGER PRIMARY KEY, position INTEGER, size INTEGER, hash TEXT)")
```

Comment:
I'm not sure we need this called each time, we can assume at this point that the table is created.

Reply:
Do you want to get rid of it or just have it run when its empty?

Reply:
I think we can get rid of this particular statement yeah

Reply:
Ok, but then it fails the test because there is no table to add to

Reply:
The table created in the test doesn't persist into the function call?

Reply:
The first test doesn't create a table

Reply:
Ah okay. Let's fix that

Reply:
Just pushed to your branch, let me know if that fixes it

Reply:
Had to push once more to remove the DB at the end of the first test apologies